bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/type/UidUtils.java

```diff
@@ -0,0 +1,217 @@
+import org.openhab.binding.siemenshvac.internal.metadata.SiemensHvacMetadataMenu;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.type.ChannelGroupTypeUID;
+import org.openhab.core.thing.type.ChannelTypeUID;
+
+/**
+ * Utility class for generating some UIDs.
+ *
+ * @author Laurent Arnal - Initial contribution
+ */
+@NonNullByDefault
+public class UidUtils {
+    /**
+     * The methods remove specific local character (like 'é'/'ê','â') so we have a correctly formated UID from a
+     * localize item label
+     *
+     * @param label
+     * @return the label without invalid character
+     */
+    public static String sanetizeId(String label) {
+        String result = label;
+
+        if (!Normalizer.isNormalized(label, Normalizer.Form.NFKD)) {
+            result = Normalizer.normalize(label, Normalizer.Form.NFKD);
+            result = result.replaceAll("\\p{M}", "");
+        }
+
+        result = result.replace(' ', '_');
+        result = result.replace(':', '_');
+        result = result.replace('.', '_');
+        result = result.replace('\'', '_');
+        result = result.replace('(', '_');
+        result = result.replace(')', '_');
+        result = result.replace('&', '_');
+        result = result.replace('/', '_');
+        result = result.replace('°', '_');
+        result = result.replace('+', '_');
+
+        return result;
+    }
+
+    /**
+     * Generates the ThingTypeUID for the given device. If it's a Homegear device, add a prefix because a Homegear
+     * device has more datapoints.
+     */
+    public static ThingTypeUID generateThingTypeUID(SiemensHvacMetadataDevice device) {
+        String type = sanetizeId(device.getType());
+        return new ThingTypeUID(SiemensHvacBindingConstants.BINDING_ID, type);
+    }
+
+    public static ThingTypeUID generateThingTypeUID(String name) {
+        return new ThingTypeUID(SiemensHvacBindingConstants.BINDING_ID, name);
+    }
+
+    /**
+     * get a more user friendly description from English short descriptor
+     *
+     * @param descriptor
+     * @return
+     */
+    private static String normalizeDescriptor(String descriptor) {
+        String result = descriptor.trim();
+
+        if (result.indexOf("CC") >= 0 || result.indexOf("HC") >= 0) {
+            for (int idx = 0; idx < 4; idx++) {
+                result = result.replace("CC" + idx, "CC");
+                result = result.replace("HC" + idx, "HC");
+            }
+        }
+
+        result = result.toLowerCase();
+
+        if (result.indexOf("history") >= 0) {
+            for (int idx = 0; idx < 20; idx++) {
+                result = result.replace("history " + idx, "history");
+            }
+        }
+
+        result = result.replace(" mon", "");
+        result = result.replace(" yue", "");
```

Comment:
Should it be:
```suggestion
        result = result.replace(" tue", "");
```

?